cpp/src/arrow/util/io_util_test.cc

```diff
@@ -522,6 +532,14 @@ TEST(CreateDirTree, Basics) {
   ASSERT_OK_AND_ASSIGN(fn, temp_dir->path().Join("EF"));
   ASSERT_OK_AND_ASSIGN(created, CreateDirTree(fn));
   ASSERT_TRUE(created);
+
+  ASSERT_OK_AND_ASSIGN(fn, temp_dir->path().Join("AB/file"));
```

Comment:
I checked and this code does the right thing with symlinks today (There is a `S_ISLNK` but you have to use `lstat` to get it) :+1: .  I'm not sure if you want to add a test case or not.

Reply:
I'd rather not complicate the tests with that (especially as it would require some symlink creation code for Windows).